src/dawn/engine_dawn.cc

```diff
@@ -471,67 +653,247 @@ Result EngineDawn::DoClear(const ClearCommand* command) {
   return map.result;
 }
 
-Result EngineDawn::DoDrawRect(const DrawRectCommand*) {
+Result EngineDawn::Helper::CreateRenderPipelineDescriptor(
+    const RenderPipelineInfo& render_pipeline,
+    const ::dawn::Device& device) {
   Result result;
 
-#if 0
-  // Add one more command to the command buffer builder, which is to
-  // copy the framebuffer texture to the framebuffer host-side buffer.
-  ::dawn::Buffer& fb_buffer = render_pipeline_info_.fb_buffer;
-  if (!fb_buffer) {
-    return Result(
-        "Dawn::DoDrawRect: Framebuffer was not created.  Did you run "
-        "any graphics pipeline commands?");
-  }
-
-  result = CreateCommandBufferBuilderIfNeeded();
+  auto* amber_format =
+      render_pipeline.pipeline->GetColorAttachments()[0].buffer->GetFormat();
+  if (!amber_format)
+    return Result("Color attachment 0 has no format!");
+  ::dawn::TextureFormat fb_format{};
+  GetDawnTextureFormat(*amber_format, &fb_format);
   if (!result.IsSuccess())
```

Comment:
Result hasn't been used yet, do we want to remove this? If so, move the Reuslt on 659 down to 676 where it's used.

Reply:
Good catch! I missed saving the outcome of GetDawnTextureFormat in _result_. Now it's being used.